src/python/module/smartredis/configoptions.py

```diff
@@ -25,20 +25,45 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import typing as t
+from functools import wraps
+from uuid import uuid4
 
 from .error import RedisRuntimeError
 from .smartredisPy import PyConfigOptions
 from .util import exception_handler, typecheck
 
-_NOT_FACTORY = (
-    "Method called on a ConfigOptions object not created from a factory method"
-)
+
+class _Managed:
+    """Marker class identifying factory-created objects"""
+
+class _FactoryMeta(type):
+    @classmethod
+    def instantiate(mcs, base: t.Type[t.Any]) -> t.Any:
+        """Factory method for creating managed instances"""
+        bases = (_Managed, base)
+        unique_key = str(uuid4()).split("-", 1)[0]
+        class_name = "".join(base.__name__ for base in bases) + unique_key
+        managed_class = mcs(class_name, bases, {})
+        return managed_class()
```

Comment:
Shifted the name generation into nested method so it could have a nice docstring & name to make that part clear.

Added a comment to explain the base class tuple